src/components/PeoplePage.tsx

```diff
@@ -1,8 +1,100 @@
+import { useEffect, useMemo, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { SortBy } from '../types/SortBy';
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(false);
+
+  const [searchParams] = useSearchParams();
+
+  useEffect(() => {
+    setIsLoading(true);
+    setError(false);
+
+    getPeople()
+      .then(setPeople)
+      .catch(() => setError(true))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const preperedPeople: Person[] = useMemo(() => {
+    let withMumDad = people.map(person => {
+      const mother = people
+        .find(mum => mum.name === person.motherName) || undefined;
```

Comment:
Redundant, because find universally returns this value if nothing is found, need to fix everywhere
```suggestion
        .find(mum => mum.name === person.motherName);
```